src/client.ts

```diff
@@ -55,8 +55,32 @@ export default class UnleashClient extends EventEmitter {
     }
   }
 
+  isParentDependencySatisfied(feature: FeatureInterface | undefined, context: Context) {
+    if (!feature?.name || !feature.dependencies?.length) {
+      return true;
+    }
+
+    return feature.dependencies.every(parent => {
+      if (this.repository.getToggle(parent.feature)?.dependencies?.length) {
+        return false;
+      }
+
+      if (parent.enabled !== false) {
+        if (parent.variants?.length) {
+          return parent.variants.includes(this.getVariant(parent.feature, context).name);
+        }
+        return this.isEnabled(parent.feature, context, () => false);
+      }
+
+      return !this.isEnabled(parent.feature, context, () => false);
+    });
+  }
+
   isEnabled(name: string, context: Context, fallback: Function): boolean {
     const feature = this.repository.getToggle(name);
+    if(!this.isParentDependencySatisfied(feature, context)) {
```

Comment:
Doesn't this guard clause mean there's no impression data for the toggle if the parent causes it to be disabled?

Reply:
Done

Reply:
added a test to make sure no one makes the same mistake again :)